src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationCore.Tests/BinaryFormat/DataObjectTests.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using PresentationCore.Tests.TestUtilities;
+using PresentationCore.Tests.FluentAssertions;
+using System.Formats.Nrbf;
+using System.Drawing;
+using System.Windows;
+using System.Threading;
+
+namespace PresentationCore.Tests.BinaryFormat;
+
+public class DataObjectTests
+{
+    [Theory]
+    [MemberData(nameof(DataObject_TestData_UnSupportedObjects))]
+    public void DataObject_UnSupportedObjects_SetData(object value)
+    {
+
+        // Create a thread to run the test logic
+        Thread testThread = new Thread(() => TestLogic(value));
+        testThread.SetApartmentState(ApartmentState.STA); // Set the thread to STA
+        testThread.Start();
+        testThread.Join(); // Wait for the thread to complete
+        
+    }
+
+    private void TestLogic(object value)
+    {
+        using BinaryFormatterScope formatterScope = new(enable: false);
+        DataObject dataObject = new();
+        dataObject.SetData(DataFormats.Serializable, value);
+
+        Clipboard.SetDataObject(dataObject, true);
+        IDataObject ClipboardDataObject = Clipboard.GetDataObject();
+        try
+        {
+            if (ClipboardDataObject != null)
+            {
+                Assert.Throws<System.Runtime.InteropServices.COMException>(() => ClipboardDataObject.GetData(DataFormats.Serializable));   
+            }
+            else
+            {
+                Assert.Fail("ClipboardDataObject is null.");
+            }
+        }
+        finally
+        {
+            Clipboard.Clear();
```

Comment:
Pass scenario should also be accounted. Also, we should output for which object failure has been observed.

Reply:
Resolving the comment as we discussed